### ISSUE TITLE
Differences between MD5 and SHA1 hash digests Endianness

### ISSUE THREAD

**Issue by jackyalbo** [intel/isa-l_crypto#57]
As part of using both MD5 and SHA1 hashes in our project it seems that there is a difference in the Endianness between them: it seems that digest words for SHA1 are big-endian, while MD5 ones are little-endian.
Is that really the case? and if so, why? as we want to use the same code for both it caused it to fail constantly for SHA1

**Comment by jackyalbo** [comment]
The changed we needed to add in our code
https://github.com/noobaa/noobaa-core/blob/540d5e587af47188b9bea84d4a0525eac87825ff/src/native/tools/ssl_napi.cpp#L137

**Comment by gbtucker** [comment]
Yes, sorry for the inconsistency but you are correct.  Looking at the code in ssl_napi.c, does hashing always go through the submit_and_flush() function?  If so, you may not be getting the possible throughput improvement if you can maintain multiple outstanding jobs.  I only looked at the `Hasher` class members though.

**Comment by guymguym** [comment]
Hey @gbtucker 
You're right - we still use a single buffer in our code.
See here for context https://github.com/intel/isa-l_crypto/issues/45#issuecomment-638277664

**Comment by gbtucker** [comment]
Oh, didn't realize they were linked.  Let me know if I can help with multi-flight.
